src/main/java/com/denizenscript/ddiscordbot/events/DiscordLeaveUserScriptEvent.java

```diff
@@ -0,0 +1,86 @@
+package com.denizenscript.ddiscordbot.events;
+
+import net.aufdemrand.denizencore.events.ScriptEvent;
+import net.aufdemrand.denizencore.objects.Element;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizencore.scripts.containers.ScriptContainer;
+import net.aufdemrand.denizencore.utilities.CoreUtilities;
+import sx.blah.discord.handle.impl.events.guild.member.UserLeaveEvent;
+
+public class DiscordLeaveUserScriptEvent extends ScriptEvent {
+    public static DiscordLeaveUserScriptEvent instance;
+
+    // <--[event]
+    // @Events
+    // discord user leaves (for <bot>)
+    //
+    // @Regex ^on discord user leaves(for [^\s]+)?$
+    //
+    // @Triggers when a Discord user leaves a guild.
+    //
+    // @Plugin dDiscordBot
+    //
+    // @Context
+    // <context.bot> returns the ID of the bot.
+    // <context.group> returns the group ID.
+    // <context.group_name> returns the group name.
+    // <context.user_id> returns the author's internal ID.
+    // <context.user_name> return's the author's name.
+    //
+    // -->
+
+    @Override
+    public boolean couldMatch(ScriptContainer scriptContainer, String s) {
+        return CoreUtilities.toLowerCase(s).startsWith("discord user leaves");
+    }
+
+    @Override
+    public boolean matches(ScriptContainer scriptContainer, String s) {
+        String lower = CoreUtilities.toLowerCase(s);
+        if (lower.equals("discord user leaves")) {
+            return true;
+        }
+        else return CoreUtilities.xthArgEquals(4, lower, botID);

```

Comment:
Only if the previous arg is `for` (to avoid conflict). Also, remove the `.equals` check above

Reply:
Not exactly sure which way you want this implemented. I have thought of using `lower.split(" ")[3].equals("for")`

Reply:
... You... are commenting on a line that shows how to check if an argument at a specific index is equal to a specific string... and chose to make up an unrelated and very bad way of doing it?

Reply:
I honestly don't get what you want me to change here. The `matches` function was copied from `DiscordMessageReceivedScriptEvent` and then IntelliJ suggested a slight optimization.

Reply:
Actually the original message received code is bork... eck